### PR TITLE
Add minor improvements to package resolution

### LINKED ISSUE #32499: BuildOptions.acceptTheirs() method mutates the current instance
The current behavior in `BuildOptions.acceptTheirs()` method does not follow package API design guidelines.  

https://github.com/ballerina-platform/ballerina-lang/blob/8c58b12b23ee4bc7b28b963e517f2eb330644c63/compiler/ballerina-lang/src/main/java/io/ballerina/projects/BuildOptions.java#L89

We need to return a new instance of the `BuildOptions` method rather than modifying the current instance

### PR DESCRIPTION
**Pull request by azinneera**
- Introduce project API test util for offline project loading
- Prioritize packages from the local repo
- Fix issue in updating Dependencies.toml
- Fixes #32499

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java [hunk 1]
```diff
@@ -37,6 +45,74 @@
...
     private static final String OS = System.getProperty("os.name").toLowerCase(Locale.getDefault());
 
+    public static BuildProject loadBuildProject(Path projectPath) {
+        BuildOptions buildOptions = new BuildOptionsBuilder().offline(true).build();
+        return BuildProject.load(projectPath, buildOptions);
+    }
+
+    static BuildProject loadBuildProject(Path projectPath, BuildOptions options) {
+        BuildOptions buildOptions = new BuildOptionsBuilder().offline(true).build();
+        options.acceptTheirs(buildOptions);
```
Comment (sameerajayasoma) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java:55]: Here we are mutating the buildOptions. Can we return a new instance with the method `options.acceptTheirs(buildOptions);`?

**Review by sameerajayasoma — APPROVED** [review]
Please address my comment with another PR.

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java (hunk 1 above)
Comment (sameerajayasoma) [reply project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java:55]: I think we need to fix this issue 
https://github.com/ballerina-platform/ballerina-lang/issues/32499

This is a fundamental issue/.

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java (hunk 1 above)
Comment (azinneera) [reply project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java]: Resolved with 5001c23

**Review by sameerajayasoma — APPROVED** [review]
I think we should update BuildOptions and CompilationOptions based on the design pattern described here. 
https://blogs.oracle.com/javamagazine/java-builder-pattern-bloch

I did a slight variation of that in ResolutionOptions. Please have a look. 

ResolutionOptions options = ResolutionOptions.builder()
                                                    .setSticky(false)
                                                    .setOffline(true)
                                                    .build();

If you agree we can update that in another PR. This is an API change though

**Pull request merged**